drivers/npmx/npmx_shell.c

```diff
@@ -1671,6 +1680,154 @@ static int cmd_ldsw_ldo_voltage_get(const struct shell *shell, size_t argc, char
 	return 0;
 }
 
+static int ldsw_soft_start_config_set(const struct shell *shell, size_t argc, char **argv,
+				      ldsw_soft_start_config_type_t config_type)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+	if (argc < 2) {
+		shell_error(shell, "Error: missing LDSW instance index and config value.");
+		return 0;
+	}
+	if (argc < 3) {
+		shell_error(shell, "Error: missing config value.");
+		return 0;
+	}
+
+	int err = 0;
+	uint8_t ldsw_indx = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT8_MAX);
+	uint32_t config_val = (uint32_t)shell_strtoul(argv[2], 0, &err);
+
+	if (err != 0) {
+		shell_error(shell, "Error: instance index and config value must be integers.");
+		return 0;
+	}
+
+	if (ldsw_indx >= NPMX_PERIPH_LDSW_COUNT) {
+		shell_error(shell, "Error: LDSW instance index is too high: no such instance.");
+		return 0;
+	}
+
+	npmx_ldsw_t *ldsw_instance = npmx_ldsw_get(npmx_instance, ldsw_indx);
+	npmx_ldsw_soft_start_config_t soft_start_config;
+	npmx_error_t err_code = npmx_ldsw_soft_start_config_get(ldsw_instance, &soft_start_config);
+
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to read soft-start config.");
+	}
+
+	char *unit = SHELL_STRING_UNIT_NONE;
+	switch (config_type) {
+	case LDSW_SOFT_START_TYPE_ENABLE:
+		soft_start_config.enable = !!config_val;
+		break;
+
+	case LDSW_SOFT_START_TYPE_CURRENT:
+		npmx_ldsw_soft_start_current_t current =
+			npmx_ldsw_soft_start_current_convert(config_val);
+
+		if (current == NPMX_LDSW_SOFT_START_CURRENT_INVALID) {
+			shell_error(shell, "Error: wrong soft-start value.");
+			return 0;
+		}
+		soft_start_config.current = current;
+		unit = SHELL_STRING_UNIT_MA;
+		break;
+	}
+
+	err_code = npmx_ldsw_soft_start_config_set(ldsw_instance, &soft_start_config);
+
+	shell_print(shell, "Success: %d.", soft_start_config.enable);
+
+	if (check_error_code(shell, err_code)) {
+		shell_print(shell, "Success: %s%s.", argv[2], unit);
```

Comment:
From what I can see, in some places `argv` value is printed after setting the value, and sometimes the value converted to an integer. I think it should be unified. @wilu-nordic @masz-nordic what do you think it should be?

Reply:
Here it can stay as is. In the rework task, it should be unified based on the confluence page. @magp-nordic could you add this case to npmx shell write instruction?

Reply:
>  @magp-nordic could you add this case to npmx shell write instruction?

Yes, I will.